components/user_settings/display/manage_languages/manage_languages.tsx

```diff
@@ -1,144 +1,176 @@
+    user: UserProfile;
+    locale: string;
+    updateSection: (section: string) => Promise<void>;
+    actions: Actions;
+};
+
+type State = {
+    isSaving: boolean;
+    openMenu: boolean;
+    locale: string;
+    serverError?: string;
+    selectedOption: object;
+};
+
+export default class ManageLanguage extends React.PureComponent<Props, State> {
+    reactSelectContainer: React.RefObject<HTMLDivElement>;
+    constructor(props: Props) {
         super(props);
-        const locales = I18n.getLanguages();
+        const locales: any = I18n.getLanguages();
         const userLocale = props.locale;
-        const selectedOption = {value: locales[userLocale].value, label: locales[userLocale].name};
+        const selectedOption = {
+            value: locales[userLocale].value,
+            label: locales[userLocale].name
+        };
         this.reactSelectContainer = React.createRef();
 
         this.state = {
             locale: props.locale,
             selectedOption,
             isSaving: false,
-            openMenu: false,
+            openMenu: false
         };
     }
 
-    componentDidMount() {
-        if (this.reactSelectContainer.current) {
-            this.reactSelectContainer.current.addEventListener('keydown', this.handleContainerKeyDown);
+    componentDidMount(this: any) {
+        const reactSelectContainer = this.reactSelectContainer.current;
+        if (reactSelectContainer) {
+            reactSelectContainer.addEventListener(
+                'keydown',
+                this.handleContainerKeyDown
+            );
         }
     }
 
-    componentWillUnmount() {
+    componentWillUnmount(this: any) {
         if (this.reactSelectContainer.current) {
-            this.reactSelectContainer.current.removeEventListener('keydown', this.handleContainerKeyDown);
+            this.reactSelectContainer.current.removeEventListener(
+                'keydown',
+                this.handleContainerKeyDown
+            );
         }
     }
 
-    handleContainerKeyDown = (e) => {
+    handleContainerKeyDown = (e: React.KeyboardEvent) => {
+        const modalBody = document.querySelector('.modal-body');
         if (isKeyPressed(e, Constants.KeyCodes.ESCAPE) && this.state.openMenu) {
-            document.querySelector('.modal-body').classList.remove('no-scroll');
+            modalBody?.classList.remove('no-scroll');
             this.setState({openMenu: false});
             e.stopPropagation();
         }
-    }
+    };
 
-    handleKeyDown = (e) => {
+    handleKeyDown = (e: React.KeyboardEvent) => {
+        const modalBody = document.querySelector('.modal-body');
         if (isKeyPressed(e, Constants.KeyCodes.ENTER)) {
-            document.querySelector('.modal-body').classList.add('no-scroll');
+            modalBody?.classList.add('no-scroll');
             this.setState({openMenu: true});
         }
-    }
+    };
 
-    setLanguage = (selectedOption) => {
+    setLanguage = (selectedOption: any) => {
```

Comment:
I think you should be able to declare the type of this. The type should be `{value: string, label: string}`

Reply:
resolved, but got an error on the onChange function on this method?